### PR TITLE
feat(ui-components) Add Dialog Component

### PR DESCRIPTION
**Pull request by robertpin**

### CODE REVIEW

File: react/features/base/ui/components/web/Dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,249 @@
+import { Theme } from '@mui/material';
+import React, { ReactElement, useCallback, useContext, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { keyframes } from 'tss-react';
+import { makeStyles } from 'tss-react/mui';
+
+import { DialogTransitionContext } from '../../../dialog/components/web/DialogTransition';
```
Comment (gabiborlea) [inline react/features/base/ui/components/web/Dialog.tsx:7]: Wouldn't be better to have this Context in this new component folder, as I think we'll replace the old dialog with this one?

Comment (robertpin) [reply react/features/base/ui/components/web/Dialog.tsx:7]: Actually, I think it's better to have the new component in the base/dialog folder so I'll move it there

Comment (saghul) [reply react/features/base/ui/components/web/Dialog.tsx]: I disagree. Having the entire UI tookit in base/ui/ makes more sense.

Comment (robertpin) [reply react/features/base/ui/components/web/Dialog.tsx]: But then what about the base/dialog folder? We'll have DialogContainer in one place and Dialog in another

Comment (robertpin) [reply react/features/base/ui/components/web/Dialog.tsx]: Discussed with Saul and we decided to move the web components to base/ui

**Review by gabiborlea — APPROVED** [review]

**Pull request merged**